### PR TITLE
Feature: Add infrastructure modules and dockerfile

### PR DESCRIPTION
**Pull request by rouderz**

### CODE REVIEW

**Review by listerineh — APPROVED** [review]
Nice :+1:

**Comment by listerineh** [comment]
Pero se cayó el check xd

**Review by listerineh — CHANGES_REQUESTED** [review]
Arregla 😾

File: api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM public.ecr.aws/lambda/python:3.9
+
+ADD ioet-secret-santa ${LAMBDA_TASK_ROOT}/ioet-secret-santa
+ADD commons ${LAMBDA_TASK_ROOT}/commons
+
+COPY requirements.txt  .
+
+RUN  pip3 install -r requirements.txt --target "${LAMBDA_TASK_ROOT}"
+
+CMD [ "ioet-secret-santa.main.handler" ]
```
Comment (Alan1108) [inline api/Dockerfile:10]:
```suggestion
CMD [ "ioet-secret-santa.main.handler" ]

```

File: infrastructure/backend/main.tf [hunk 2]
```diff
@@ -17,3 +17,164 @@ terraform {
...
+    }
+  }
+
+  allow_overwrite = true
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = 60
+  type            = each.value.type
+  zone_id         = data.aws_route53_zone.secret-santa_zone.zone_id
+}
```
Comment (Alan1108) [inline infrastructure/backend/main.tf:113]: can you take all this code from the module aws-subdomain-acm

File: infrastructure/backend/main.tf [hunk 3]
```diff
@@ -17,3 +17,164 @@ terraform {
...
+
+resource "aws_acm_certificate_validation" "acm_cert_validation" {
+  depends_on = [
+    data.aws_route53_zone.secret-santa_zone,
+    aws_route53_record.secret-santa_record
+  ]
+  certificate_arn         = aws_acm_certificate.secret-santa_cert.arn
+  validation_record_fqdns = [for record in aws_route53_record.secret-santa_record : record.fqdn]
+}
```
Comment (Alan1108) [inline infrastructure/backend/main.tf:134]: this one also

**Review by Alan1108 — CHANGES_REQUESTED** [review]

**Pull request merged**